### PR TITLE
Split analysis

### PR DESCRIPTION
**Pull request by tomer-algo**
# Intro
  - Split Analysis to SmS and General images
# Project
  - Suggested Skeleton
# Implemented Sections
  - Split Analysis to SmS and General images
# Tests Description
  - 
# Undeveloped Sections
  - 
# What's New
  - 
# Checklist
  - [x] Module tested fully after last commit
  - [ ] Scheduled a PR meeting (if needed)
  - [ ] Documentation (comments or a dedicated documnet)
### Tested platforms
  - [x] Local
  - [ ] Cloud (Sagemaker)
  - [ ] Staging
  - [ ] Not relevant
### Tested Cameras
  - [ ] Gen 1.5
  - [ ] Gen 2
  - [x] Not relevant
# Comments
### Included Features
  - 
### Infrastructure Dependencies
  - 
### Tech Comments
  - 
### Testing Instructions
  - 
### Knowledge
| / | Link |
| ------ | ------ |
| Monday |  |
| TLD |  |
| Tech Spec |  |
### Todos
  - What is left to do/test

### CODE REVIEW

File: analysis/analysis_metrics.py [hunk 1]
```diff
@@ -106,6 +108,40 @@ def check_if_hit_and_update(self, distance, head):
...
+        self.total_count = 0
+
+    def update_counter(self, pckh_data):
+        self.hit_count += pckh_data.hit_count
+        self.total_count += pckh_data.total_count
+
+
+class PCKHAnalysisCounters:
+    def __init__(self):
+        self.all_joints_0_5 = PCKHCounter()
```
Comment (Nimrods) [inline analysis/analysis_metrics.py:123]: Remind me why not putting all those "all_joints_0_x" in a dictionary so it won't be hard-coded and we can change it easily in the future

File: analysis/analysis_metrics.py [hunk 2]
```diff
@@ -106,6 +108,40 @@ def check_if_hit_and_update(self, distance, head):
...
+        self.all_joints_0_1 = PCKHCounter()
+        self.all_joints_0_1_on_general_images = PCKHCounter()
+        self.all_joints_0_1_on_sms_images = PCKHCounter()
+
+        self.sms_joints_0_1 = PCKHCounter()
+        self.sms_joints_0_1_on_general_images = PCKHCounter()
+        self.sms_joints_0_1_on_sms_images = PCKHCounter()
+
+    def update_counters(self, joint_name, joint_values, sms_joints):
+        self.all_joints_0_5.update_counter(joint_values.pckh_0_5)
```
Comment (Nimrods) [inline analysis/analysis_metrics.py:134]: I think it's not a scalable code
for now it's ok, but if we'll have more values or more partitions it won't be good enough
Consider change to structure and iterate over the counters

Comment (tomer-algo) [reply analysis/analysis_metrics.py:134]: in that case we will need the mapping between the counters and the relevant joints_values member

File: analysis/analysis_metrics.py (hunk 1 above)
Comment (tomer-algo) [reply analysis/analysis_metrics.py:123]: same comment as below
we will also need to have a mapping between the counter to the JointsAnalysis class member

File: analysis/analysis_metrics.py (hunk 2 above)
Comment (Nimrods) [reply analysis/analysis_metrics.py:134]: Not sure I understand...
What is the problem in keeping the counters inside a dictionary that its keys and values will reflect the hierarchy here, something like:
`['all_joints']['0.5']=PCKHCounter()`

So afterwards you can iterate over the keys and values and do whatever you want

File: analysis/analysis_metrics.py (hunk 1 above)
Comment (Nimrods) [reply analysis/analysis_metrics.py:123]: moving the conversation to the comment below

File: analysis/analysis_metrics.py (hunk 2 above)
Comment (tomer-algo) [reply analysis/analysis_metrics.py:134]: Each `joint_values` is a `JointsAnalysis` class 
so when updating the counters we need a map between the `PCKHAnalysisCounters` members to the relevant members in `JointsAnalysis`

Comment (tomer-algo) [reply analysis/analysis_metrics.py:134]: Lets do short conversation

**Pull request merged**